google/cloud/storage/internal/throw_status_delegate.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/throw_status_delegate.h"
+#include "google/cloud/terminate_handler.h"
+#include <sstream>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+[[noreturn]] void ThrowStatus(Status&& status) {
```

Comment:
Is there a reason for this signature to require a `Status` rvalue instead of accepting a `Status` value and letting the caller decide if they want to move or copy the argument into this function?

Reply:
No good reason. Fixed.